test/functional/import-abort-rescan.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test wallet import RPCs.
+
+Test rescan behavior of importprivkey when aborted. The test ensures that:
+1. The abortrescan command indeed stops the rescan process.
+2. Subsequent rescan catches the aborted address UTXO
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (assert_equal, get_rpc_proxy)
+from decimal import Decimal
+import threading # for bg importprivkey
+import time      # for sleep
+
+class ImportAbortRescanTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        # self.setup_clean_chain = False
+
+    def run_test(self):
+        # Generate for BTC
+        self.nodes[0].generate(300)
+        self.log.info("Creating blocks with transactions ...")
+        # Make blocks with spam to cause rescan delay
+        for i in range(200):
+            if i % 50 == 0: self.log.info("... %2.f%%", 100.0 / 200.0 * i)
+            addr = self.nodes[0].getnewaddress()
+            for j in range(10):
+                self.nodes[0].sendtoaddress(addr, 0.1)
+            self.nodes[0].generate(1)
+        self.log.info("Sending to shared address ...")
+        addr = self.nodes[0].getnewaddress()
+        privkey = self.nodes[0].dumpprivkey(addr)
+        self.nodes[0].sendtoaddress(addr, 0.123)
+        self.nodes[0].generate(10) # mature tx
+        self.sync_all()
+        balances = [n.getbalance() for n in self.nodes]
+
+        # Import this address in the background ...
+        self.log.info("Importing address in background thread ...")
+        node1ref = get_rpc_proxy(self.nodes[1].url, 1, timeout=600)
+        importthread = threading.Thread(target=node1ref.importprivkey, args=[privkey])
+        importthread.start()
+        # ... then abort rescan; try a bunch until abortres becomes true,
+        # because we will start checking before above thread starts processing
+        self.log.info("Attempting to abort scan ...")
+        for i in range(2000):
+            time.sleep(0.001)
+            abortres = self.nodes[1].abortrescan()
+            if abortres: break
+        assert abortres # if false, we failed to abort
+        self.log.info("Waiting for import thread to die ...")
+        # import should die soon
+        for i in range(10):
+            time.sleep(0.1)
+            deadres = not importthread.isAlive()
+            if deadres: break
+
+        assert deadres # if false, importthread did not die soon enough
```

Comment:
nit: prefer `assert deadres, "importthread did not die soon enough"` (so the error message is delivered to the user)